lib/storage.dart

```diff
@@ -189,168 +59,29 @@ class Storage {
-    }
-
-    //generate list of activities
-    List<int> pickRand = List<int>.generate(ActivityName.values.length-1, (i)=>i+1);
-    pickRand.shuffle();
-    List<ActivityName> activities = List<ActivityName>.generate(3, (int index)=>ActivityName.values[pickRand[index]], growable: false);
-    await _db.insert(_dailyResetTable, <String, Object>{
-      'date': DateTime.now().daysSinceEpoch(),
-      'activity_1_id':await getActivityId(activities[0]),
-      'activity_2_id':await getActivityId(activities[1]),
-      'activity_3_id':await getActivityId(activities[2])
-    });
-    return List<Map<String,Object>>.generate(3, (int index)=>{'activity':activities[index],'completed':false});
-  }
-
-  Future<void> setDailyCompleted(int activityNumber) async{
-    var row = (await _db.query(_dailyResetTable,columns: ['activity_completed','id'], where: 'date == ?', whereArgs: [DateTime.now().daysSinceEpoch()], limit: 1))[0];
-    int completionInfo = row['activity_completed'] as int;
-    completionInfo |= (1 << (activityNumber-1));
-    await _db.update(_dailyResetTable, {'activity_completed': completionInfo}, where: 'id == ?', whereArgs: [row['id']]);
+  /// Inserts a new mood journal entry
+  Future<void> insertMoodJournal(
+      String date, String mood, int intensity, String note) async {
+    await _db.insert(
+      _moodJournalsTable,
+      {
+        'date': date,
+        'mood': mood,
+        'intensity': intensity,
+        'note': note,
+      },
+    );
   }
 
-  /// Retrieves all cues for a specific session ID
-Future<List<Map<String, dynamic>>> getCuesForSession(int sessionId) async {
-  return await _db.query(
-    'cues',
-    where: 'session_id = ?',
-    whereArgs: [sessionId],
-  );
-}
-/// Inserts a new cue for a specific session
-Future<void> insertCue(int sessionId, int timeSec, String message) async {
-  await _db.insert(
-    'cues',
-    {
-      'session_id': sessionId,
-      'time_sec': timeSec,
-      'message': message,
-    },
-  );
-}
-/// Inserts a new session into the sessions table
-Future<void> insertSession(int sessionId, String name) async {
-  await _db.insert(
-    'sessions',
-    {
-      'id': sessionId,
-      'name': name,
-    },
-    conflictAlgorithm: ConflictAlgorithm.ignore,  // Ignores if session already exists
-  );
-}
-
-
-  /// For testing only
-  Future<void> deletePreference(PreferenceName preference) async {
-    await _db.delete(_preferencesTable, where: 'name = ?', whereArgs: [preference.name]);
+  /// Retrieves all mood journal entries from the database
+  Future<List<Map<String, dynamic>>> getAllMoodJournalEntries() async {
+    return await _db.query(
+      _moodJournalsTable,
+      orderBy: 'date DESC',
```

Comment:
The dates are stored as strings. Does this sorting work?

This is why I've been storing dates as the days since the Unix epoch since that can be stored as an integer and is sortable.